File: exercism_data/src/183.rb
class Phrase

  def initialize phrase
    words = phrase.downcase.split(/[\W_]+/)
    @values = words.inject({}) do |h, word|
      h[word] ? h[word] += 1 : h[word] = 1
      h
    end
  end

  def word_count
    @values
  end

end

File: assignments/python/wc/src/183.py
import re
def word_count(string):
	wc = {}
	str = re.sub('\W+',' ', string)
	words = str.lower().split()
	for word in words:
		if wc.has_key(word):
			wc[word] += 1
		else:
			wc[word] = 1
	return wc
